accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,320 @@
+
+```json
+{
+	"inventory-label": "my-ansible-control-node",
+        "hosts": [
+		"my-ansible-managed-system1.foo.bar",
+		"my-ansible-managed-system2.foo.bar",
+		"my-ansible-managed-system3.foo.bar"
+	],
+	"local_stored_inventory": "/var/lib/spacewalk/ansible/inventory-label/inventory.yaml",
+	"optional_local_stored_playbooks": [
+		"/var/lib/spacewalk/ansible/inventory-label/playbooks/playbook1.yaml",
+		"/var/lib/spacewalk/ansible/inventory-label/playbooks/playbook2.yaml",
+		"/var/lib/spacewalk/ansible/inventory-label/playbooks/playbook3.yaml"
+	]
+}
+```
+
+As you can see, besides of reporting the hosts from the Ansible inventory, the idea is that we also stored it locally in the Uyuni server in order to being able to reuse it later to target those hosts with "salt-ssh" directly from the Uyuni server.
+
+In short, we would gather, for instance:
+- Ansible inventory (hosts)
+- Ansible SSH keys referenced in the inventory
+- Playbooks under specified "remote playbook path" (to be displayed in the UI - readonly mode - not editable)
+
+This would be stored under `/var/lib/spacewalk/ansible/inventory-label/` path in the Uyuni server. At the time of processing the inventory file by "ansible-gatherer", it needs to be tailored to adapt the path of the SSH keys to the local path in the Uyuni server. As done in script from Ansible Integration PoC [here](https://github.com/meaksh/uyuni-hacks/blob/master/scripts/ansible/import_systems_from_ansible_controller.py).
+
+With this information, Java is able to proceed creating these systems in Uyuni, as "Foreign / ANSIBLE" entitled systems.
+
+- If any of those systems is already existing with "ANSIBLE" entitlement, then nothing to do
+- If the system is already registered but not with "ANSIBLE" entitlement, then add "ANSIBLE" entitlement.
+
+Of course, we need some new tables in the database, at least something like:
+
+```
+"suseAnsibleController" (label, type, org_1)
+"suseAnsibleControllerConfig" (control_node_id, parameter, value)
+"suseAnsibleControllerSystem" (system_id, control_node_id)
+"suseAnsibleControllerPlaybook" (control_node_id, local_path, content?)
+```
+
+And also create the new "ANSIBLE" entitlement, which should be an addon-entitlement compatible with Salt, Foreign and maybe also for a potential "Management/ANSIBLE" (traditional client) entitlement. Here is the [SQL script](https://github.com/meaksh/uyuni-hacks/blob/master/scripts/ansible/add_ansible_entitleme.sql) used for PoC demo. It's also needed to adapt the rhn databaste functions implemented in the DB.
+
+NOTE: The profile for a system with registered in Uyuni as "Foreign/ANSIBLE" does not look like as the one Salt/Traditional system, since most of the Uyuni features (channels, software profile, etc) are not implemented for neither "Foreign" nor "ANSIBLE" entitlements. In order to get the entire portfolio of Uyuni features for Salt minion, this "Foreign/ANSIBLE" systems would need to be transitioned to "Salt/ANSIBLE" minion.
```

Comment:
Observation: "entitlements" have been renamed to "system types" in all user-facing material (including the UI).

References to "entitlements" still exist in the codebase, but I would recommend to stick to "system types" here, as "entitlements" can be a bit confusing as a concept.

Reply:
Thanks for the note. Fixed.